methcomp/blandaltman.py

```diff
@@ -48,16 +50,29 @@ def __init__(
 
         # variables assignment
         self.computed = False
-        self.method1: np.array = np.asarray(method1)
-        self.method2: np.array = np.asarray(method2)
         self.diff_method: str = diff
         self.CI: float = CI
         self.loa: float = limit_of_agreement
+        super().__init__(method1, method2)
+
+    def _check_params(self):
+        super()._check_params()
+
+        if len(self.method1) != len(self.method2):
+            raise ValueError("Length of method 1 and method 2 are not equal.")
 
-        # check provided parameters
-        self._check_params()
+        if self.CI is not None and (self.CI > 1 or self.CI < 0):
+            raise ValueError("Confidence interval must be between 0 and 1.")
 
-    def compute(self) -> Dict[str, Any]:
+        if self.diff_method not in ["absolute", "percentage"]:
+            raise ValueError(
+                "The provided difference method must be either absolute or percentage."
+            )
+
+        # if any([not isinstance(x, str) for x in [self.x_title, self.y_title]]):
```

Comment:
Should probably delete commented out block

Reply:
resolved